01_financial-well-being/income_metrics_all.qmd

```diff
@@ -0,0 +1,676 @@
+puma_place_2021 <- puma_place_2021 %>% 
+  select(statefip, puma, place, pop20, afact, afact2)
+
+```
+
+Drop observations where the weight adjustment is zero.
+
+```{r}
+puma_place_2021 <- puma_place_2021 %>%
+  filter(afact!= 0.000)
+```
+
+Create a variable that assigns a weight to each place based on the total 2020 population. First, create a variable for the total population.
+
+```{r}
+puma_place_2021 <- puma_place_2021 %>%
+  mutate(totpop = sum(pop20))
+
+```
+
+Then, create a variable for the share of total population comprised by each place (unique state fip + county pairs)
+
+```{r}
+puma_place_2021 <- puma_place_2021 %>% 
+  group_by(statefip, place) %>% 
+  mutate(placepop = sum(pop20),
+                placewgt = placepop/totpop)
+
+```
+
+Look at distribution of place total population shares (unique). NYC, LA and Chicago all hold disproportionate shares of the population.
+
+```{r}
+
+puma_place_2021 %>%
+  select(statefip, place, placewgt) %>%
+  unique() %>%
+  ggplot(aes(x = placewgt)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Place pop weight") +
+  ylab("Density")
+
+
+```
+
+### Create quality variable
+
+Prepare for Data Quality flag (at end).
+
+Create flags in the PUMA-place crosswalk for high percentage of data from outside of place. Per Greg (for counties), 75% or more from the county - in this case, place - is good, below 35% is bad, in between is marginal. This is calculated by taking the product of percentage of PUMA in place and percentage of place in PUMA for each place-PUMA pairing, and summing across the place.
+
+Create new vars of interest.Products is the sum of the share of the place captured in the PUMA (afact) and the share of the PUMA captured in the place (afact2).
+
+```{r}
+
+puma_place <- puma_place_2021 %>%
+  mutate(products = afact*afact2)
+
+puma_place <- puma_place %>%
+  group_by(statefip, place) %>%
+  mutate(sum_products = sum(products),
+                place_pop = sum(pop20))
+
+
+```
+
+Sum_products mean is 0.76.
+Bottom percentile of population starts at 110629.
+
+```{r}
+summary(puma_place)
+```
+
+```{r}
+
+puma_place <- puma_place %>%
+  mutate(
+    puma_flag = 
+      case_when(
+        sum_products >= 0.75 ~ 1,
+        sum_products >= 0.35 ~ 2,
+        sum_products < 0.35 ~ 3
+      ),
+    small_place = 
+      case_when(
+        placepop >= 110629 ~ 0,
+        placepop < 110629 ~ 1
+      )
+  )
+
+```
+
+Save as "puma_place.csv" in the intermediate file.
+
+```{r}
+write_csv(puma_place, "data/intermediate/puma_place.csv")
+
+```
+
+Save a version with just the place-level values of data quality variables
+
+```{r}
+
+place_puma <- puma_place %>%
+  group_by(statefip, place) %>% 
+  summarise(puma_flag = mean(puma_flag), 
+            small_place = mean(small_place))
+
+```
+
+Save as "place_puma.csv" in the intermediate file.
+
+```{r}
+write_csv(place_puma, "data/intermediate/place_puma.csv")
+```
+
+### Prepare microdata
```

Comment:
This jumps around a little bit. I would move all of the crosswalk work before loading the IPUMS data instead of loading IPUMS data, then working with the crosswalk, then prepping the IPUMS data.